engine/apps/grafana_plugin/tasks/sync.py

```diff
@@ -139,3 +141,26 @@ def sync_team_members_for_organization_async(organization_pk):
 
     grafana_api_client = GrafanaAPIClient(api_url=organization.grafana_url, api_token=organization.api_token)
     sync_team_members(grafana_api_client, organization)
+
+
+@shared_dedicated_queue_retry_task(autoretry_for=(Exception,), max_retries=1)
+def cleanup_empty_deleted_integrations(organization_pk, dry_run=True):
+    try:
+        organization = Organization.objects.get(pk=organization_pk)
+    except Organization.DoesNotExist:
+        logger.info(f"Organization {organization_pk} was not found")
+        return
+
+    integrations_qs = (
+        AlertReceiveChannel.objects_with_deleted.filter(deleted_at__isnull=False, organization=organization)
+        .annotate(num_alert_groups=Count("alert_groups"))
```

Comment:
nit: consider using `Exists` instead of `Count`, it could be faster for orgs that have a lot of alert groups: `from django.db.models import Exists`

or this (same logic but the query is simpler):

```python3
AlertReceiveChannel.objects.filter(organization=organization, deleted_at__isnull=True, alert_groups=None).distinct()
```